### DISCUSSION TITLE
[Feature] Support for multiple pages from the same repo

### DISCUSSION THREAD

**Discussion by felhag** [community/community#47867] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Hey,

I use github pages for some of my projects which works really well. One my repositories is creating two artifacts which I want to deploy to two different domains. The only way I can achieve this now is by duplicating my repository. I think it would be a great feature for Github Pages if we can deploy multiple branches to multiples pages. For example:

![image](https://user-images.githubusercontent.com/62592273/219857259-8ef0d9ab-25f8-4052-8b45-e101ec29015c.png)

Note: This feature was already proposed in 2020 with [this ticket](https://github.com/github/pages-gem/issues/724). @yoannchaudet advised to open an issue in this community, so here we are :)

**Comment by codebydant** [comment]
This will be really important for monorepos where we have docs folders in different locations in the repository. We will be able to choose each path in the main branch to deploy a readthedocs github page per subproject.

**Comment by bethac07** [comment]
Another thing this would be really nice for - multilingual sites! Right now, we have content we can publish to different branches for different languages, but can only serve one as a "live" page.

**Reply by TheQueenIsDead** [reply]
That would require that you merge singular changes to common components across multiple branches. Imagine trying to update your `favicon.ico` with 25 internationalisations. 25 pull requests. 
 
It would likely be wise to use i18n functionality that is already provided, for example:
 - Hugo https://gohugo.io/functions/i18n/
 - Vue https://www.npmjs.com/package/vue-i18n
 - Angular https://angular.io/guide/i18n-overview
 
Most frameworks have the ability to do this for you

**Reply by bethac07** [reply]
Most do, but JupyterBook does not have first-class support for this, at least not yet! This is a good workaround until it does, as long as the number of languages is, as you point out, not huge

**Comment by nicfv** [comment]
> I think it would be a great feature for Github Pages if we can deploy multiple branches to multiples pages.

I like this. It should even be generalized to supporting multiple pages in a single branch. For example if I have my main app in `dist/` and documentation in `docs/` (in the same branch), I may want to publish the app to `example.com` and documentation to `docs.example.com`. I could see this work well as a setting in the [Deploy Github Pages](https://github.com/marketplace/actions/deploy-github-pages-site) action.

**Comment by rngadam** [comment]
In addition to deploying to different domains, this feature would also helpful if we need to have versions at different stages of development (dev, uat, prod).

**Comment by what1s1ove** [comment]
Something similar... but native support would be more desirable!
https://github.com/marketplace/actions/deploy-static-app-in-branch-to-gh-pages

**Comment by SergeyRoyt** [comment]
Found the solution for Github pages with multiple subfolders here
https://github.com/github/docs/discussions/26012#discussioncomment-6203134

Single branch, multiple directories with different websites with URL route navigation between them - good enough for my case

**Reply by nicfv** [reply]
This is not bad, and definitely works for some implementations, but it's got a few limitations.
1. This forces you to have everything in the same branch, which is one of the issues brought up in this thread.
2. To me this feels more like a single website rather than 3 separate ones. I don't think doing it this way will allow you to assign a custom domain for each of these subdirectories, so you're stuck with this structure: `user-name.github.io/repo/<website-name>`

Either way, thank you for sharing this solution!

**Reply by what1s1ove** [reply]
Yeah... the main problem is attaching the domain

**Reply by nauderuan** [reply]
Broken link?

**Reply by TWiStErRob** [reply]
@nauderuan not the original but probably along these lines: https://stackoverflow.com/a/41879114/253468, same solution as described by @SergeyRoyt below the link.

**Comment by Ratstail91** [comment]
I would very much like this!

My current project is undergoing a rewrite. I'd like to be able to store the v1, v2, v1-docs and v2-docs all in one repo, with the docs deployed at the websites `toylang.com` and `v2.toylang.com`. This will let me swap them around at the DNS level when the v2 docs are ready.

As it is, the v2 docs seem to require a separate repo. Thanks!

**Comment by svathyitis** [comment]
Is this now a feature or still in bottom pipeline?

**Comment by svathyitis** [comment]
Not sure, why such a basic feature is missed! Imagine having different sub-domains and each would need a separate repo. Not a scalable option.

**Reply by Ratstail91** [reply]
It's definitely awkward, I worked around it by mirroring the repo, and adding a git push to the CI. Here it is in practice:

* https://github.com/krgamestudios/Toy/tree/v1-docs/.github/workflows
* https://github.com/krgamestudios/toylang-mirror

In a nutshell, `pages-mirror.yml` triggers on a push to the branch `v1-docs`, pushing to the mirror repo using the branch name `mirror`. Then, `pages-deploy.yml` triggers when a branch called `mirror` is pushed, and handles the actual publishing of the webpage.

Where there's a will there's a way. hope that helps you!
